src/recipeUtils.js

```diff
@@ -1,15 +1,62 @@
-export function recipesFromTag(recipes, tag) {
+import recipeData from '../src/data/recipes-sample.js';
+import ingredientsData from '../src/data/ingredients-sample.js';
+
+
+function recipesFromTag(recipes, tag) {
   return recipes.filter((recipe) => {
     return recipe.tags.includes(tag)
   });
 }
 
-export function recipesfromName(recipes, name) {
+function recipesfromName(recipes, name) {
   return recipes.filter((recipe) => {
     return recipe.name.includes(name)
   })
 }
 
+const findRecipe = (recipeData, recipeName) => {
+  const recipe = recipeData.find(({ name }) => name === recipeName);
+  return recipe
+};  
+
+const findIngredientNames = (recipeData, recipeName) => {
+  const recipe = findRecipe(recipeData, recipeName);
+    if (recipe === undefined) {
+      return "Sorry, we don't have that recipe."
+    }
+  const ingredientIds = recipe.ingredients.map(({ id }) => id);
+  const ingredients = ingredientsData.filter(({ id }) => ingredientIds.includes(id));
+  return ingredients.map(({ name }) => name);
+};
+
+const calculateRecipeCost = recipe => {
+  if (recipe === undefined) {
+    return 'Sorry, that recipe cannot be found.'
+  }
+  const totalCost = recipe.ingredients.reduce((acc, {id, quantity: {amount}}) => 
+    acc + (amount / 100) * (ingredientsData.find(data => data.id === id).estimatedCostInCents), 0)
+  return totalCost.toFixed(2);
+}
+
+  const recipeInstructions = recipe => {
+   if (recipe === undefined) {
+     return "Sorry, that recipe cannot be found.";
+   }
+   const instructions = recipe.instructions.map(({ number, instruction }) => `${number}. ${instruction}`);
+   return instructions.join(' ')
+ };
+
+/// still trying to figure out a way to put each instruction on a new line... my new line of thinking is using two prototype.methods... .join then .string ??  -- let me know what you think...
+// -- Andrea
```

Comment:
Love the way this function is formatted! Not sure how to put each instruction on a new line, but I think that is something we can figure out together

Reply:
I hope so! Thanks Denzel!